mapper/orm/src/main/java/org/hibernate/search/mapper/orm/mapping/impl/HibernateOrmTypeContextContainer.java

```diff
@@ -99,6 +107,12 @@ private HibernateOrmTypeContextContainer(Builder builder, SessionFactoryImplemen
 			byJpaEntityNameContent.put( typeContext.jpaEntityName(), typeContext );
 
 			byHibernateOrmEntityNameContent.put( typeContext.hibernateOrmEntityName(), typeContext );
+
+			for ( PojoRawTypeIdentifier<?> superType : typeContext.ascendingSuperTypes() ) {
+				if ( isManagedType( sessionFactory, superType.javaClass() ) ) {
+					indexedWithSuperTypesByExactClassContent.put( superType.javaClass(), superType );
```

Comment:
So... not necessarily indexed then? This adds contained types.

Maybe rename that stuff to `allIncludingSuperTypesByExactClassContent`, or something like that?